rgw/v2/lib/decorators.py

```diff
@@ -18,16 +18,27 @@ def check_pem(function):
     @wraps(function)
     def wrapper(*args, **kwargs):
         log.info("kwargs: {}".format(kwargs))
+        log.info("args: {}".format(args[2]))
         log.info("executing func: {name}".format(name=function.__name__))
         log.info("ssl: {}".format(kwargs.get("ssl")))
+        ssh_con = args[2]
         if kwargs.get("ssl"):
-            log.info("in decorator: checking if pem file exists ")
-            if not pem.check_pem_file_exists():
-                log.info("pem file does not exist")
-                log.info("creating pem file")
-                pem.create_pem()
+            if ssh_con:
```

Comment:
Can we do something like below:
if kwargs.get("ssl"):
if ssh_con:
ssh_con = ssh_con
else:
ssh_con = None

log.info("in decorator: checking if pem file exists ")
if not pem.check_pem_file_exists(ssh_con):
       log.info("pem file does not exist")
       log.info("creating pem file")
       pem.create_pem(ssh_con)
else:
       log.info("in decorator: pem file already exists")

Reply:
Made changes in code